JamesLiuZX

JamesLiuZX opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” 

– David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines 
- [x]  Managing events

For launching it yourself for experimental purposes,
Run the following class under Launcher.java:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class,args);
    }
}
```
`Lets go`

Commit:
Add Gradle support

JamesLiuZX committed:
Implemented Level-1 Functionality: Greet, Echo, Exit

JamesLiuZX committed:
Merge branch 'master' of github.com:JamesLiuZX/ip

# Conflicts:
#	src/main/java/Duke.java

JamesLiuZX committed:
Level-2 Implementation: Add, List

JamesLiuZX committed:
Level-3

JamesLiuZX committed:
Level-4

JamesLiuZX committed:
A-TextUiTesting

JamesLiuZX committed:
Level 5

JamesLiuZX committed:
Level-6

JamesLiuZX committed:
Improved logic flow and error handling for inputs

JamesLiuZX committed:
Level-7 Save

JamesLiuZX committed:
Merge pull request #1 from JamesLiuZX/Level-7

Level-7

JamesLiuZX committed:
Level-8

JamesLiuZX committed:
A-MoreOOP

JamesLiuZX committed:
Merge conflicts

JamesLiuZX committed:
Merge pull request #2 from JamesLiuZX/Level-8

Level-8

bojie3 commented on src/main/java/TaskEvent.java:
```diff
@@ -0,0 +1,34 @@
+import java.time.LocalDateTime;
```
Might be better to put import statement below class docs to standardize across all

JamesLiuZX replied:
Noted.

bojie3 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,125 @@
+        records = new ArrayList<>(size);
+    }
+
+    public void mark(int index) {
```
Remember to add javadocs for all the public function

JamesLiuZX replied:
Noted.

bojie3 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    public static String format(String input) {
+        return "____________________________________________________________\n" +
```
Can define the string as the private variable to make code cleaner

JamesLiuZX replied:
Good suggestion, thanks

bojie3 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,44 @@
+                }
+                String[] inputArr = input.split(" ", 2);
+                //Check for invalid commands (first word)
+                boolean validCommand = commandList.contains(inputArr[0]);
```
Can edit boolean variable to sound more like boolean variable. Eg isValidcomment

bojie3 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,69 @@
+                String status = split[1];
+                String taskName = split[2];
+                index++;
+                switch (cmd) {
```
Nice use of switch statements

JamesLiuZX replied:
Thanks!

bojie3 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,89 @@
+        if (input.contains("/")) {
+            String[] split = input.split("/", 2);
+            String[] secondSplit = split[0].split(" ", 2);
+
```
Can remove this empty line, makes code looks nicer imo

JamesLiuZX replied:
Noted.

bojie3 reviewed (commented):
Overall code is easy to understand and good use of java oop concepts.

JamesLiuZX committed:
A-Packages

JamesLiuZX committed:
Add minor reformatting based on peer review

JamesLiuZX committed:
Merge branch 'add-gradle-support'

mayuanxin1234 commented on src/main/java/duke/exceptions/InvalidCommandException.java:
```diff
@@ -0,0 +1,10 @@
+
+import duke.functions.Ui;
+
+public class InvalidCommandException extends DukeException{
```
Good expansion of exceptions to consider more exceptions

JamesLiuZX replied:
Implemented more, thanks!

mayuanxin1234 commented on src/main/java/duke/functions/Parser.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    public void handleTwoInputs(String command, Integer index, TaskList d) {
+        switch (command) {
```
Good use of switch to make code look cleaner

mayuanxin1234 commented on src/main/java/duke/functions/Storage.java:
```diff
@@ -0,0 +1,73 @@
+            Scanner fileReader = new Scanner(f);
+            System.out.println("Database loaded.");
+            int index = -1;
+            while (fileReader.hasNextLine()) {
```
Can put this under Parser class

JamesLiuZX replied:
Did it, thanks!

mayuanxin1234 commented on src/main/java/duke/functions/Ui.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.Scanner;
+
+public class Ui {
+    public static ArrayList<String> commandList = new ArrayList<>(Arrays.asList
```
This can be implemented with enums

mayuanxin1234 reviewed (commented):
Good code with some improvement advice.

Commit:
A-JUnit

Commit:
A-Jar

Commit:
A-JavaDoc

Commit:
A-CodingStandard

Commit:
Level-9

JamesLiuZX committed:
Merge pull request #3 from JamesLiuZX/branch-A-JavaDoc

Add JavaDocs to all class methods.

JamesLiuZX committed:
Merge branch 'master' into branch-Level-9

JamesLiuZX committed:
Merge branch 'master' into branch-A-CodingStandard

JamesLiuZX committed:
Merge pull request #5 from JamesLiuZX/branch-A-CodingStandard

Merge branch-A-CodingStandard

JamesLiuZX committed:
Merge pull request #4 from JamesLiuZX/branch-Level-9

Merge branch-Level-9

ryanchua00 commented on src/main/java/duke/exceptions/DateTimeFormatException.java:
```diff
@@ -0,0 +1,17 @@
+ * @author JamesLiuZX
+ * AY2223-S2 CS2103T
+ */
+public class DateTimeFormatException extends DukeException {
+    /**
+     * The default constructor for this exception.
+     */
+    public DateTimeFormatException() {
+        super("Please input the datetime format in /by DD/MM/YY HHMM.");
+    }
```
I love this class to make throwing exceptions so much easier.

ryanchua00 commented on src/main/java/duke/functions/Ui.java:
```diff
@@ -0,0 +1,70 @@
+                }
+                String[] inputArr = input.split(" ", 2);
+                //Check for invalid commands (first word)
+                boolean isValidCommand = commandList.contains(inputArr[0]);
```
Clever way of detecting valid input

ryanchua00 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,239 @@
+     * Inserts a TaskDeadline from the database to the TaskList.
+     *
+     * @param name Task name to be added.
+     * @param time Deadline time in string, according to the dateTimeRegex format.
+     */
+    public void insertDeadline(String name, String time) throws DateTimeFormatException {
+        System.out.println(time);
+        try {
+            if (!time.matches(dateTimeRegex)) { // dd/mm/yy tttt
+                throw new DateTimeFormatException();
+            }
+            LocalDateTime dateTime = LocalDateTime.parse(time, formatter);
+            TaskDeadline d = new TaskDeadline(time, dateTime);
+            this.records.add(d);
+            dw.addToDb(d);
+            System.out.println(Ui.format("Got it. I've added this task:\n" + d.toString()));
+        } catch (DateTimeFormatException e) {
+            System.out.println(e.getMessage());
+        }
```
Would it be better to throw the Exception and catch it where the method is called?

ryanchua00 reviewed (commented):
Great, readable code. Everything is done with clear intent. Many try-catch blocks are made within sub-classes, wondering if it would be easier to handle those exceptions program flow methods. i.e ui.run()

JamesLiuZX committed:
Level-10

JamesLiuZX committed:
A-CheckStyle

JamesLiuZX committed:
Fix checkstyle

JamesLiuZX committed:
A-Assertions

Add assertions in key points in code which can make or break the program flow. The points added are:
* Checking for null TaskList during reading of database
* Checking for null FILEPATH
* Checking for null Duke object in MainWindow

JamesLiuZX committed:
BCD-Extension

Add an extension in the form of a help page where users can type
* 'help' to gain a full list of commands available,
* 'usage' to understand the format and purpose of each command.

JamesLiuZX committed:
Add Github CI .yml file

JamesLiuZX committed:
A-UserGuide

Update README.md to according to the recommended formats of a user guide.

JamesLiuZX committed:
Set jekyll theme

JamesLiuZX committed:
Update user photo

JamesLiuZX committed:
Update README.md User Guide

JamesLiuZX committed:
A-BetterGui

First draft of the Johnny GUI

JamesLiuZX committed:
Remove IO redirection from gradle.yml workflow.

JamesLiuZX committed:
Add minor README Fix

A dummy commit to merge branch Level-10 to the master branch to fulfil the IP requirements as the branching was forgotten previously.

JamesLiuZX committed:
Level-10 Merge

A dummy commit to merge branch Level-10 to the master branch to fulfil the IP requirements as the branching was forgotten previously.

JamesLiuZX committed:
A-Assertions

JamesLiuZX committed:
Add A-Assertions Update

Dummy branch to account from forgetting to add A-Assertions as a branch before incrementing.

JamesLiuZX committed:
Add UI fix

JamesLiuZX committed:
Add bugfixes for Parser class

JamesLiuZX committed:
Add checkstyle fixes

JamesLiuZX committed:
Add checkstyle fixes

JamesLiuZX committed:
Add Ui.png

JamesLiuZX committed:
Finalize GUI border padding and color

JamesLiuZX committed:
Finalize User Guide

JamesLiuZX committed:
A-Release

JamesLiuZX committed:
Edit User guide

JamesLiuZX committed:
Edit User guide

JamesLiuZX committed:
Add final docs changes
